src/app/components/core/developer/additional-services/additional-service-modal/additional-service-modal.component.ts

```diff
@@ -298,21 +321,29 @@ export class AdditionalServiceModalComponent {
     if (name != null) {
       serviceName = name;
     }
-    this.toggleModal(false, {service: serviceName, state: 'enable', isCancelEvent: false});
+    this.additionalServicesUtils.enableService(serviceName, this.fromNavbar);
+    // added some delay to close the modal because 
+    // enabling service takes time to get the updated state from API
+    setTimeout(() => {
+      this.toggleModal(false);             
```

Comment:
We should carefully use the set timeout because the time given is not a guarantee that page will be available in that estimated time.